module/x/gravity/keeper/msg_server_test.go

```diff
@@ -0,0 +1,68 @@
+package keeper
+
+import (
+	"encoding/hex"
+	"strings"
+	"testing"
+
+	"github.com/Gravity-Bridge/Gravity-Bridge/module/x/gravity/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestConfirmHandlerCommon(t *testing.T) {
+	input, ctx := SetupFiveValChain(t)
+	defer func() { input.Context.Logger().Info("Asserting invariants at test end"); input.AssertInvariants() }()
+
+	privKey, err := crypto.GenerateKey()
+	require.NoError(t, err)
+
+	ethAddress, err := types.NewEthAddress(crypto.PubkeyToAddress(privKey.PublicKey).String())
+	require.NoError(t, err)
+
+	input.GravityKeeper.SetEthAddressForValidator(ctx, ValAddrs[0], *ethAddress)
+	input.GravityKeeper.SetOrchestratorValidator(ctx, ValAddrs[0], AccAddrs[0])
+
+	batch := types.OutgoingTxBatch{
+		TokenContract: "0xd041c41EA1bf0F006ADBb6d2c9ef9D425dE5eaD7",
+		BatchTimeout:  420,
+	}
+
+	checkpoint := batch.GetCheckpoint(input.GravityKeeper.GetGravityID(ctx))
+
+	ethSignature, err := types.NewEthereumSignature(checkpoint, privKey)
+	require.NoError(t, err)
+
+	sv := msgServer{input.GravityKeeper}
+	err = sv.confirmHandlerCommon(input.Context, ethAddress.GetAddress().Hex(), AccAddrs[0], hex.EncodeToString(ethSignature), checkpoint)
+	assert.Nil(t, err)
+}
+
+func TestConfirmHandlerCommonWithLowercaseAddress(t *testing.T) {
```

Comment:
we should also try all uppercase addresses and incorrect non-checksum mixed capitalization addresses.

Reply:
Done!